libnavigation-base/src/main/java/com/mapbox/navigation/base/options/NavigationOptions.kt

```diff
@@ -59,6 +63,7 @@ data class NavigationOptions(
         private var isFromNavigationUi: Boolean = false
         private var isDebugLoggingEnabled: Boolean = false
         private var deviceProfile: DeviceProfile = HandheldProfile()
+        private var rerouteAdapter: RouteRefreshAdapter = RouteRefreshAdapterProvider.provideRouteRefreshAdapter()
```

Comment:
we don't need RouteRefreshAdapterProvider as an object

but a factory provider like this could be a good way to allow clients to provide their own implementation of RouteRefresh. If we were to give customers an option to provide their own implementation, provider could be a good place to do it. 

But it can't be overridden as an `object`, so there's no benefit making RouteRefreshAdapterProvider an `object`